### PR TITLE
fix: make bug tests pass and flag issues for todos

### PR DESCRIPTION
**Pull request by willbouch**

### CODE REVIEW

File: test/todos.test.js [hunk 1]
```diff
@@ -442,16 +438,13 @@ describe('Test for todos endpoints', function () {
...
-        expect((await chai.request(host).get('/todos')).body.todos.length).equal(defaultTodosObject.todos.length);
-
-        // FAILURE - Title is not supposed to be mandatory
+        expect(res).to.have.status(400);
+        
+        /* FAILURE - Title is not supposed to be mandatory
+        From the documentation, the body should only contain properties that we want to amend. So, not putting
+        the title is supposed to be fine. However, it returns an error saying that the title is a mandatory field
+        even it if is not. Should return a 200.
+        */
```
Comment (Louismousine) [inline test/todos.test.js:447]: the documentation on ghub says that PUT replaces the whole object, but yeah it aint in the jar documentation so i guess it's fine

Comment (willbouch) [reply test/todos.test.js:447]: good catch. I think we should focus on the api doc given thouth

File: test/todos.test.js [hunk 2]
```diff
@@ -563,10 +550,13 @@ describe('Test for todos endpoints', function () {
...
         const id = 123;
         const res = await chai.request(host).get(`/todos/${id}/tasksof`);
-        expect(res).to.have.status(404);
-        expect(res.body).to.deep.equal({ errorMessages: [`Could not find related projects for instance with todos/${id}`] });
+        expect(res).to.have.status(200);
 
-        // FAILURE - Should not be returning any information except an error message
+        /* FAILURE - Should not be returning any information except an error message
+        Getting the projects of an unexisting todo should throw an error. If this is not a bug, this is a bad
+        design decision since the user should be informed that the id given in the request corresponds to no
```
Comment (Louismousine) [inline test/todos.test.js:557]: didnt this return error messages??? pretty sure it did when i wrote it

File: test/todos.test.js [hunk 3]
```diff
@@ -644,9 +639,11 @@ describe('Test for todos endpoints', function () {
         await chai.request(host).delete(`/todos/${todoId}/tasksof`);
         const postRes = await chai.request(host).post(`/todos/${todoId}/tasksof`).send(body);
         expect(postRes).to.have.status(400);
-        expect(postRes.body).to.deep.equal({ errorMessages: [`java.lang.NullPointerException`] });
 
-        // FAILURE - The response should be a message, not a java exception.
+        /* FAILURE - The response should be a message, not a java exception.
+        The system should handle the error when an unexpected property is sent in the body. There should't be internal
```
Comment (Louismousine) [inline test/todos.test.js]: should(n)'t

File: test/todos.test.js (hunk 2 above)
Comment (willbouch) [reply test/todos.test.js:557]: no it doesnt

File: test/todos.test.js [hunk 4]
```diff
@@ -644,9 +639,11 @@ describe('Test for todos endpoints', function () {
         await chai.request(host).delete(`/todos/${todoId}/tasksof`);
         const postRes = await chai.request(host).post(`/todos/${todoId}/tasksof`).send(body);
         expect(postRes).to.have.status(400);
-        expect(postRes.body).to.deep.equal({ errorMessages: [`java.lang.NullPointerException`] });
```
Comment (Louismousine) [inline test/todos.test.js:647]: keep this

Comment (willbouch) [reply test/todos.test.js:647]: Well it shoudnt send this. I can add it in the comment that this is what we receive, but i dont think i want to keep an assertion for that

**Pull request merged**